loader/src/packfile_io.rs

```diff
@@ -8,42 +8,56 @@ use std::{
 use capnp::serialize::SliceSegments;
 use distill_core::{utils::make_array, AssetMetadata, AssetRef, AssetUuid};
 use distill_schema::pack::pack_file;
-use memmap::{Mmap, MmapOptions};
 use thread_local::ThreadLocal;
 
+#[cfg(not(target_arch = "wasm32"))]
+use memmap::{Mmap, MmapOptions};
+
 use crate::{
     io::{DataRequest, LoaderIO, MetadataRequest, MetadataRequestResult, ResolveRequest},
     loader::LoaderState,
 };
 
-struct PackfileMessageReader {
+trait PackfileMessageReader: Send + Sync {
+    fn get_reader(&self) -> capnp::Result<pack_file::Reader<'_>>;
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+struct PackfileMessageReaderFile {
     file: ManuallyDrop<File>,
     mmap: ManuallyDrop<Mmap>,
     message_reader: ManuallyDrop<ThreadLocal<capnp::message::Reader<SliceSegments<'static>>>>,
 }
-impl PackfileMessageReader {
+
+#[cfg(not(target_arch = "wasm32"))]
+impl PackfileMessageReaderFile {
     pub fn new(file: File) -> std::io::Result<Self> {
         let mmap = unsafe { MmapOptions::new().map(&file)? };
-        Ok(PackfileMessageReader {
+        Ok(PackfileMessageReaderFile {
             file: ManuallyDrop::new(file),
             mmap: ManuallyDrop::new(mmap),
             message_reader: ManuallyDrop::new(ThreadLocal::new()),
         })
     }
+}
 
+#[cfg(not(target_arch = "wasm32"))]
+impl PackfileMessageReader for PackfileMessageReaderFile {
     fn get_reader(&self) -> capnp::Result<pack_file::Reader<'_>> {
-        let messge_reader = self.message_reader.get_or_try(|| {
+        let message_reader = self.message_reader.get_or_try(|| {
             // We ensure that the reader is dropped before the mmap so it's ok to cast to 'static here
             let slice: &[u8] = &self.mmap;
             let mut slice: &[u8] = unsafe { std::mem::transmute::<&[u8], &'static [u8]>(slice) };
```

Comment:
Should probably put the comment into the `unsafe` block and prefix it with `SAFETY: `.